tests/tests/test_decommission.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+# Copyright 2016 Mender Software AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from fabric.api import *
+import pytest
+from common import *
+from common_setup import *
+from helpers import Helpers
+from common_update import common_update_procedure
+from MenderAPI import inv, adm, deploy, deviceauth
+from mendertesting import MenderTesting
+
+# ugly way of passing a the "true" parameter to "standard_setup_one_client"
+@pytest.mark.parametrize("standard_setup_one_client", ["force_new"], indirect=True)
+class TestDeviceDecommissioning(MenderTesting):
+
+
+    @MenderTesting.fast
+    def test_device_decommissioning(self, standard_setup_one_client):
+        """ Decommission a device successfully """
+
+        if not env.host_string:
+            execute(self.test_device_decommissioning, standard_setup_one_client, hosts=get_mender_clients())
+            return
+
+        adm.check_expected_status("pending", len(get_mender_clients()))
+        adm_id = adm.get_devices()[0]["id"]
+        device_id = adm.get_devices()[0]["device_id"]
+
+        # decommission device before it's accepted results in 404
+        deviceauth.decommission(adm_id, expected_http_code=404)
```

Comment:
IMO the status should be 200 as long as there was a single request from the device. If it's not then that's a bug in deviceauth